sormas-backend/src/main/java/de/symeda/sormas/backend/common/AbstractCoreFacadeEjb.java

```diff
@@ -0,0 +1,161 @@
+
+	public AbstractCoreFacadeEjb() {
+	}
+
+	public AbstractCoreFacadeEjb(Class<ADO> adoClass, Class<DTO> dtoClass, SRV service, UserService userService) {
+		super(adoClass, dtoClass, service, userService);
+	}
+
+	@Override
+	public DTO getByUuid(String uuid) {
+		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
+		return convertToDto(service.getByUuid(uuid), pseudonymizer);
+	}
+
+	@Override
+	public List<DTO> getByUuids(List<String> uuids) {
+		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
+		return service.getByUuids(uuids).stream().map(c -> convertToDto(c, pseudonymizer)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<DTO> getAllAfter(Date date) {
+		return getAllAfter(date, null, null);
+	}
+
+	public List<DTO> getAllAfter(Date date, Integer batchSize, String lastSynchronizedUuid) {
+		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
+		return service.getAllAfter(date, batchSize, lastSynchronizedUuid)
+			.stream()
+			.map(c -> convertToDto(c, pseudonymizer))
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public DTO save(@Valid @NotNull DTO dto) {
+		ADO existingAdo = dto.getUuid() != null ? service.getByUuid(dto.getUuid()) : null;
+		DTO existingDto = toDto(existingAdo);
+
+		Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
+		restorePseudonymizedDto(dto, existingDto, existingAdo, pseudonymizer);
+
+		validate(dto);
+
+		existingAdo = fillOrBuildEntity(dto, existingAdo, true);
+		service.ensurePersisted(existingAdo);
+
+		return convertToDto(existingAdo, pseudonymizer);
+	}
+
+	public boolean exists(String uuid) {
+		return service.exists(uuid);
+	}
+
+	@Override
+	public void archive(String uuid) {
+		ADO ado = service.getByUuid(uuid);
+		if (ado != null) {
+			ado.setArchived(true);
+			service.ensurePersisted(ado);
+		}
+	}
+
+	public void dearchive(String uuid) {
+		ADO ado = service.getByUuid(uuid);
+		if (ado != null) {
+			ado.setArchived(false);
+			service.ensurePersisted(ado);
+		}
+	}
+
+	public boolean isArchived(String uuid) {
+		return service.isArchived(uuid);
+	}
+
+	public DTO convertToDto(ADO source, Pseudonymizer pseudonymizer) {
+
+		DTO dto = toDto(source);
+		pseudonymizeDto(source, dto, pseudonymizer);
+		return dto;
+	}
+
+	public void executeAutomaticDeletion(DeletionConfiguration entityConfig) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<ADO> cq = cb.createQuery(adoClass);
+		Root<ADO> from = cq.from(adoClass);
+
+		Date referenceDeletionDate = DateHelper.subtractDays(new Date(), entityConfig.getDeletionPeriod());
+		cq.where(cb.lessThanOrEqualTo(from.get(getDeleteReferenceField(entityConfig.getDeletionReference())), referenceDeletionDate));
+
+		List<ADO> toDeleteEntities = QueryHelper.getResultList(em, cq, null, null);
+
+		toDeleteEntities.forEach(entity -> {
+			delete(entity);
+		});
+	}
+
+	protected void delete(ADO entity) {
```

Comment:
I understand that this is not strictly needed to put it to the facade, but somehow, it looks inconsistent to me to have `save(DTO dto)` and not `delete(DTO dto)`.